sdk/python/requirements.in

```diff
@@ -34,5 +34,5 @@ absl-py>=0.9,<=0.11
 kfp-pipeline-spec>=0.1.13,<0.2.0
 fire>=0.3.1,<1
 pydantic>=1.8.2,<2
-dataclasses>=0.8,<1; python_version<"3.7"
+dataclasses>=0.8,<1; python_version<"3.9"
```

Comment:
This is a conditional dependency, it means only install `dataclasses` when Python version is less than 3.7. Because in Python 3.7 and above, `dataclasses` is in system libs. Given we no longer support any version below 3.7, this dependency can be dropped.

https://www.python.org/dev/peps/pep-0508/#environment-markers